cmd/ingester/app/consumer/consumer.go

```diff
@@ -109,8 +110,12 @@ func (c *Consumer) Close() error {
 
 func (c *Consumer) handleMessages(pc sc.PartitionConsumer) {
 	c.logger.Info("Starting message handler", zap.Int32("partition", pc.Partition()))
-	c.partitionsHeld.Inc(1)
-	defer c.partitionsHeld.Inc(-1)
+	c.partitionsHeld++
```

Comment:
Wouldn't this require some sort of synchronization? L92 starts `handleMessages` as a goroutine

Reply:
You're probably right, so I moved `partitionMapLock.Lock()` around to encompass this.

Reply:
seems like `c.partitionMapLock.Lock()` should be enough